core/src/main/scala/org/bitcoins/core/script/ScriptProgram.scala

```diff
@@ -135,6 +138,14 @@ sealed trait StartedScriptProgram extends ScriptProgram
   * evaluated by the [[org.bitcoins.core.script.interpreter.ScriptInterpreter ScriptInterpreter]].
   *
   * @param lastCodeSeparator The index of the last [[org.bitcoins.core.script.crypto.OP_CODESEPARATOR OP_CODESEPARATOR]]
+  * @param trueCount The depth of OP_IFs/OP_NOTIFs we've entered on the true condition before the first false.
+  * @param falseAndIgnoreCount The depth of OP_IFs/OP_NOTIFs we've entered after and including the first false condition.
```

Comment:
Is there any invariants we can add around this? I think I get what it is suppose to do conceptually, but it seems like every time we encounter a `OP_IF/NOT_IF` one of these numbers should get incremented?